src/client/QXmppTransferManager.cpp

```diff
@@ -426,7 +426,7 @@ void QXmppTransferJob::terminate(QXmppTransferJob::Error cause)
     d->state = FinishedState;
 
     // close IO device
-    if (d->iodevice)
+    if (d->iodevice && d->iodevice->parent() == this)
```

Comment:
I agree with Georg, I'd rather see an explicit boolean saying "we created this device"